src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,63 @@
+import { deleteTodo } from '../../api/todos';
+import { Errors } from '../../enums/Errors';
+import { useTodosContext } from '../../hooks/useTodosContext';
+import { TodoFilter } from '../TodoFilter';
+
+export const Footer: React.FC = () => {
+  const {
+    todos,
+    setTodos,
+    setLoadingTodoIds,
+    showError,
+    setIsFocusedInput,
+    notCompletedCount,
+    completedCount,
+  } = useTodosContext();
+
+  const handleDeleteTodo = async (todoId: number): Promise<void> => {
+    try {
+      await deleteTodo(todoId);
+      setTodos(prevTodos => prevTodos.filter(t => t.id !== todoId));
+      setLoadingTodoIds([]);
```

Comment:
- Avoid one-letter variables/arguments
- The same, remove loaders in finallyprevLoadingTodosIds
- And also, what if you have other loaders at the same moment?
`setLoadingTodoIds(prevLoadingTodosIds => prevLoadingTodosIds.filter(id => id !== todoId))`